### PR TITLE
Make red-hat job not dependent on tag version of build

### PR DESCRIPTION
**Pull request by pebrc**
Take the version of the release branch `VERSION` instead of the `BUILDKITE_TAG` variable to allow releasing the OLM bits with fixes that came after the release version was tagged.

### CODE REVIEW

File: .buildkite/scripts/release/redhat-preflight.sh [hunk 1]
```diff
@@ -20,7 +22,8 @@ container_already_verified() {
 }
 
 main() {
-    local tag="${1#v}"
+    local tag
+    tag=$(cat "$ROOT"/VERSION)
```
Comment (thbkrkr) [inline .buildkite/scripts/release/redhat-preflight.sh:26]: It's not you, but reading this code again, I don't like how I wrote it.
Your change will work, but the code suggests that `tag` is local but it still used by the other function `container_already_verified`.

Maybe remove the local work and use uppercase to suggest it's global? Rename it to TAG or ECK_VERSION?
```bash
    ECK_VERSION=$(cat "$ROOT/VERSION")
    export ECK_VERSION
```

Or don't use export but declare the vars outside main:

```
API_KEY=
PROJECT_ID=
ECK_VERSION=
```

**Review by thbkrkr — APPROVED** [review]
LGTM

Left a comment to improve the script a bit. I'll let you consider if it's worth it.

**Comment by pebrc** [comment]
@thbkrkr I applied your suggestion

**Comment by pebrc** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.9|[](https://github.com/elastic/cloud-on-k8s/pull/7062)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**